source/extensions/filters/network/mysql_proxy/mysql_filter.cc

```diff
@@ -0,0 +1,146 @@
+#include "extensions/filters/network/mysql_proxy/mysql_filter.h"
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/assert.h"
+#include "common/common/logger.h"
+
+#include "extensions/filters/network/well_known_names.h"
+
+#include "include/sqlparser/SQLParser.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace MySQLProxy {
+
+MySQLFilterConfig::MySQLFilterConfig(const std::string& stat_prefix, Stats::Scope& scope)
+    : scope_(scope), stat_prefix_(stat_prefix), stats_(generateStats(stat_prefix, scope)) {}
+
+MySQLFilter::MySQLFilter(MySQLFilterConfigSharedPtr config) : config_(std::move(config)) {}
+
+void MySQLFilter::initializeReadFilterCallbacks(Network::ReadFilterCallbacks& callbacks) {
+  read_callbacks_ = &callbacks;
+}
+
+Network::FilterStatus MySQLFilter::onData(Buffer::Instance& data, bool) {
+  doDecode(data);
+  return Network::FilterStatus::Continue;
+}
+
+Network::FilterStatus MySQLFilter::onWrite(Buffer::Instance& data, bool) {
+  doDecode(data);
+  return Network::FilterStatus::Continue;
+}
+
+void MySQLFilter::doDecode(Buffer::Instance& buffer) {
+  // Safety measure just to make sure that if we have a decoding error we keep going and lose stats.
+  // This can be removed once we are more confident of this code.
+  if (!sniffing_) {
+    buffer.drain(buffer.length());
```

Comment:
Hi @rshriram. Why does the buffer need draining here ? 
I tried out the filter with SSL and it hangs after draining for both `onData` and `onWrite`.
Taking out this line altogether results in a build that works as intended.

Reply:
I was merely following the logic in mongo filter, that did the same.

```
void ProxyFilter::doDecode(Buffer::Instance& buffer) {
  if (!sniffing_ ||
      !runtime_.snapshot().featureEnabled(MongoRuntimeConfig::get().ProxyEnabled, 100)) {
    // Safety measure just to make sure that if we have a decoding error we keep going and lose
    // stats. This can be removed once we are more confident of this code.
    buffer.drain(buffer.length());
    return;
  }
```